src/App.tsx

```diff
@@ -1,26 +1,259 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { addTodo, deleteTodo, getTodos, updatedTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import cn from 'classnames';
+import { Filter } from './types/types';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [inputText, setInputText] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodos, setTempTodos] = useState<Todo[]>([]);
+  const [loadingError, setLoadingError] = useState('');
+  const [validationError, setValidationError] = useState('');
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [filterSelected, setFilterSelected] = useState<Filter>(Filter.All);
+
+  const filterTodos = useMemo(() => {
+    switch (filterSelected) {
+      case Filter.Active:
+        return todos.filter(todo => !todo.completed);
+
+      case Filter.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [todos, filterSelected]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setLoadingError('Unable to load todos');
+        setTimeout(() => {
+          setLoadingError('');
+        }, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos, loadingTodos]);
+
+  const toggleTodo = (id: number) => {
+    setTodos(prevTodos =>
+      prevTodos.map(todo =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  };
+
+  const toggleTodoAll = () => {
+    const allCompleted = todos.every(todo => todo.completed);
+
+    setTodos(prevTodos =>
+      prevTodos.map(todo => ({ ...todo, completed: !allCompleted })),
+    );
+  };
+
+  const handleAddTodo = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!inputText.trim()) {
+      setValidationError('Title should not be empty');
+      setTimeout(() => {
+        setValidationError('');
+      }, 3000);
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+
+      return;
+    }
+
+    const tempTodo: Todo = {
+      id: Date.now(),
+      title: inputText,
+      completed: false,
+      userId: 1129,
```

Comment:
import USER_ID and use here

Reply:
Create a const for userId and use it here